src/libraries/Common/src/System/IO/Compression/ZLibNative.cs

```diff
@@ -317,7 +318,7 @@ public unsafe ErrorCode InflateEnd()
             }
 
             // This can work even after XxflateEnd().
-            public string GetErrorMessage() => _zStream.msg != ZNullPtr ? Marshal.PtrToStringUTF8(_zStream.msg)! : string.Empty;
+            public unsafe string GetErrorMessage() => _zStream.msg != ZNullPtr ? Utf8StringMarshaller.ConvertToManaged((byte*)_zStream.msg)! : string.Empty;
```

Comment:
```suggestion
            public unsafe string GetErrorMessage() => _zStream.msg != ZNullPtr ? Utf8StringMarshaller.ConvertToManaged(_zStream.msg)! : string.Empty;
```
And change ZStream.msg type to byte*